terraform/environments/nomis/ec2-jumpserver.tf

```diff
@@ -43,41 +43,90 @@ data "template_file" "user_data" {
   }
 }
 
-resource "aws_instance" "jumpserver_windows" {
-  instance_type               = "t3.medium"
-  ami                         = data.aws_ami.jumpserver_image.id
-  associate_public_ip_address = false
-  iam_instance_profile        = aws_iam_instance_profile.ec2_jumpserver_profile.id
-  ebs_optimized               = true
-  #checkov:skip=CKV_AWS_126: "Ensure that detailed monitoring is enabled for EC2 instances" don't think we need such fine resolution for JS
-  monitoring                  = false
-  vpc_security_group_ids      = [aws_security_group.jumpserver-windows.id]
-  subnet_id                   = data.aws_subnet.private_az_a.id
-  key_name                    = aws_key_pair.ec2-user.key_name
-  user_data                   = data.template_file.user_data.rendered
-  user_data_replace_on_change = true
+# instance launch template
+resource "aws_launch_template" "this" {
+  image_id                             = data.aws_ami.this.id
+  instance_initiated_shutdown_behavior = "terminate"
+  instance_type                        = "t3.medium"
+  key_name                             = aws_key_pair.ec2-user.key_name
+  iam_instance_profile {
+    arn = aws_iam_instance_profile.this.arn
+  }
+  block_device_mappings {
+    device_name = "jumpserver-root"
+    ebs {
+      delete_on_termination = true
+      encrypted             = true
+      volume_type           = "gp3"
+      volume_size           = 40
+    }
+  }
   metadata_options {
     http_endpoint = "enabled"
     http_tokens   = "required"
   }
-  root_block_device {
-    delete_on_termination = true
-    encrypted             = true
-    volume_type           = "gp3"
+
+  monitoring {
+    enabled = false
   }
 
-  tags = merge(
-    local.tags,
-    {
-      Name          = "jumpserver_windows"
-      os_type       = "Windows"
-      os_version    = "2022"
-      always_on     = "false"
-      "Patch Group" = aws_ssm_patch_group.windows.patch_group
-    }
-  )
+  network_interfaces {
+    associate_public_ip_address = false
+    security_groups             = [aws_security_group.jumpserver-windows.id]
+    delete_on_termination       = true
+  }
+
+  user_data = base64encode(data.template_file.user_data.rendered)
+  tag_specifications {
+    resource_type = "instance"
+    tags = merge(
+      local.tags,
+      {
+        Name          = "jumpserver_windows"
+        os_type       = "Windows"
+        os_version    = "2022"
+        always_on     = "false"
+        "Patch Group" = aws_ssm_patch_group.windows.patch_group
+      }
+    )
+  }
 }
 
+# autoscaling
+resource "aws_autoscaling_group" "this" {
+  launch_template {
+    id      = aws_launch_template.this.id
+    version = "$Default"
+  }
+  desired_capacity    = 1
+  name                = "jumpserver-autoscaling-group"
+  min_size            = 1
+  max_size            = 1
+  force_delete        = true
+  vpc_zone_identifier = [data.aws_subnet.this.id]
```

Comment:
Better to do this across multiple AZs. `private_subnets_a` and `private_subnets_b`

Reply:
Roger dodger.